apps/designer/app/shared/db/assets.server.ts

```diff
@@ -15,11 +16,18 @@ export const loadByProject = async (projectId?: Project["id"]) => {
 
 export const create = async (
   projectId: Project["id"],
-  values: { type: string; name: string; path: string }
+  values: { name: string; path: string; size: number; arrayBuffer: Uint8Array }
 ) => {
+  const image = sharp(values.arrayBuffer);
+  const metadata = await image.metadata();
```

Comment:
still neeed to measure this, I don't trust it until I see some numbers

Reply:
https://sharp.pixelplumbing.com/api-input

It actually only uses the header of the image for the metadata

Reply:
The 30 per second is opening an image, resizing it and compressing it

Reply:
https://sharp.pixelplumbing.com/performance#running-the-benchmark-test

I guess you can run it for metadata